src/operation/get-parameters-as-json-schema.ts

```diff
@@ -189,118 +203,146 @@ export default function getParametersAsJsonSchema(operation: Operation, api: OAS
-                  globalDefaults,
-                  refLogger,
-                }),
-                // Note: this applies a $schema version to each field in the larger schema object. It's not really *correct*
-                // but it's what we have to do because there's a chance that the end user has indicated the schemas are different
-                $schema: getSchemaVersionString(currentSchema, api),
-              };
             }
           }
-        }
 
-        // Parameter descriptions don't exist in `current.schema` so `constructSchema` will never have access to it.
-        if (current.description) {
-          schema.description = current.description;
-        }
+          // Parameter descriptions don't exist in `current.schema` so `constructSchema` will never have access to it.
+          if (current.description) {
+            schema.description = current.description;
+          }
 
-        prev[current.name] = schema;
+          prev[current.name] = schema;
 
-        if (current.required) {
-          required.push(current.name);
-        }
+          if (current.required) {
+            required.push(current.name);
+          }
 
-        return prev;
-      }, {});
-
-      // This typing is technically WRONG :( but it's the best we can do for now.
-      const schema: OpenAPIV3_1.SchemaObject = {
-        type: 'object',
-        properties: properties as Record<string, OpenAPIV3_1.SchemaObject>,
-        required,
-      };
-
-      return {
-        type,
-        label: types[type],
-        schema,
-        deprecatedProps: getDeprecated(schema, type),
-      };
-    });
+          return prev;
+        }, {});
+
+        // This typing is technically WRONG :( but it's the best we can do for now.
+        const schema: OpenAPIV3_1.SchemaObject = {
+          type: 'object',
+          properties: properties as Record<string, OpenAPIV3_1.SchemaObject>,
+          required,
+        };
+
+        return {
+          type,
+          label: types[type],
+          schema,
+          deprecatedProps: getDeprecated(schema, type),
+        };
+      })
+      .filter(Boolean);
+
+    if (!opts.mergeIntoBodyAndMetadata) {
+      return transformed;
+    }
+
+    // If we want to merge parameters into a single metadata entry then we need to pull all
+    // available schemas and `deprecatedProps` (if we don't want to retain them via the
+    // `retainDeprecatedProps` option) under one roof.
+    const deprecatedProps = transformed.map(r => r.deprecatedProps?.schema || null).filter(Boolean);
+    return [
+      {
+        type: 'metadata',
+        label: types.metadata,
+        schema: {
+          allOf: transformed.map(r => r.schema),
+        } as SchemaObject,
+        deprecatedProps: deprecatedProps.length
+          ? {
+              type: 'metadata',
+              schema: {
+                allOf: deprecatedProps,
+              } as SchemaObject,
+            }
+          : null,
```

Comment:
having absolutely zero knowledge of the implications of how this is all consumed, what's the reasoning behind having the fallback set to `null` vs `undefined`?

Reply:
I forget why we did this, but I agree that it should be `undefined`. Will look into it.

Reply:
Actually I'm going to leave this as `null` because we end up deleting `deprecatedProps` out of the object if it's falsy here: https://github.com/readmeio/oas/blob/main/src%2Foperation%2Fget-parameters-as-json-schema.ts#L328-L329

Changing these returns to be `undefined` is going to require the `getDeprecated()` method to have some `return undefined` calls which will look weird (and I don't want to suppress the `consistent-return` ESLint rule).